mkdocs_awesome_pages_plugin/navigation.py

```diff
@@ -248,3 +282,36 @@ def get_by_type(nav, T):
         if item.children:
             ret.extend(get_by_type(item.children, T))
     return ret
+
+
+# Copy of mkdocs.structure.pages.Page._set_title and Page.read_source
```

Comment:
Is this the only way to implement title sorting? I'm a bit worried about:
1. Having to update this as MkDocs changes its implementation
2. Potential incompatibilities with other plugins running after this one

Reply:
> Is this the only way to implement title sorting? I'm a bit worried...

I'm not sure, which part is worrying you 🤔, so I'm assuming based on the later points that you see an issue with modifying the internal MkDocs `page.title` directly using this copied external function, and I guess your idea of making it safer is to not do that, and store the titles internally in the plugin?

I wanted to make sure that the build process won't redo the same tasks, this includes the setting of the title. By default the title gathering is linked to the `page_read_source` event, which happens much later in the build process. 
My opinion was that storing the title internally is not better, because it would lead to the title being reset twice. 
However, the file is read twice either way, because the `page_read_source` event still triggers later, but the `set_title` function quits early, so I can understand the counter-point that "setting the title isn't such an expensive operation to redo".

---

> 1. Having to update this as MkDocs changes its implementation

Very unlikely, since the implementation is the same since 5 years:
https://github.com/mkdocs/mkdocs/blob/34ef3ca6d0390959080ce93a695361eea1649272/mkdocs/structure/pages.py#L147
and the project is seen as feature complete:
https://github.com/mkdocs/mkdocs/discussions/3145

Also any implementation change in MkDocs could influence a custom implementation, because I feel that even a custom implementation should abide by some "rules" that are defined in their current implementation, so I don't think there is a way to completely disconnect from MkDocs.

---

> 2. Potential incompatibilities with other plugins running after this one

That's a valid concern, if other plugins rely on the `page.title` being `None` during the `page_read_source` it might break them, but in my opinion that's a non-existent edge-case🤔
Personally I had to modify the `i18n` plugin in my fork, but that's not the edge-case above, because it reordered the nav using the awesome-pages plugin before localizing the titles 🥶. Once this PR is merged I will open a PR there.
https://github.com/kamilkrzyskow/i18n/commit/e6e44c2a49457db23ee1a8f99271e08985513634
Based on this example, I do think that other plugins might rely on awesome-pages to order the `nav` in the `on_nav` event.
Also it might be wise to update this plugin with event priorities, but I'm not sure how to do that with support for MkDocs 1.2.3 🤔 

---

Anyways, for now I'll think about where to store the internal titles... 🤔